spring-cloud-gcp-data-spanner/src/test/java/org/springframework/cloud/gcp/data/spanner/repository/SpannerRepositoryImplTests.java

```diff
@@ -0,0 +1,187 @@
+		Object ob2 = new Object();
+		Iterable<Object> ret = new SpannerRepositoryImpl(operations, Object.class)
+				.saveAll(Arrays.asList(ob, ob2));
+		assertThat(ret, containsInAnyOrder(ob, ob2));
+		verify(operations, times(1)).upsert(eq(ob));
+		verify(operations, times(1)).upsert(eq(ob2));
+	}
+
+	@Test
+	public void findByIdTest() {
+		SpannerOperations operations = mock(SpannerOperations.class);
+		String key = "key";
+		Object ret = new Object();
+		when(operations.find(eq(Object.class), eq(Key.of(key)))).thenReturn(ret);
+		assertEquals(ret,
+				new SpannerRepositoryImpl(operations, Object.class).findById(key).get());
+		verify(operations, times(1)).find(eq(Object.class), eq(Key.of(key)));
+	}
+
+	@Test
+	public void existsByIdTestFound() {
+		SpannerOperations operations = mock(SpannerOperations.class);
+		String key = "key";
+		Object ret = new Object();
+		when(operations.find(eq(Object.class), eq(Key.of(key)))).thenReturn(ret);
+		assertTrue(new SpannerRepositoryImpl(operations, Object.class).existsById(key));
+	}
+
+	@Test
+	public void existsByIdTestNotFound() {
+		SpannerOperations operations = mock(SpannerOperations.class);
+		when(operations.find(eq(Object.class), any())).thenReturn(null);
+		assertFalse(
+				new SpannerRepositoryImpl(operations, Object.class).existsById("key"));
+	}
+
+	@Test
+	public void findAllTest() {
+		SpannerOperations operations = mock(SpannerOperations.class);
+		new SpannerRepositoryImpl(operations, Object.class).findAll();
+		verify(operations, times(1)).findAll(eq(Object.class));
+	}
+
+	@Test
+	public void findAllByIdTest() {
+		SpannerOperations operations = mock(SpannerOperations.class);
+		when(operations.find(eq(Object.class), any())).thenAnswer(invocation -> {
+			KeySet keys = invocation.getArgument(1);
+			assertThat(keys.getKeys(), containsInAnyOrder("key1", "key2"));
+			return null;
+		});
+		new SpannerRepositoryImpl(operations, Object.class)
+				.findAllById(Arrays.asList("key1", "key2"));
+	}
+
+	@Test
+	public void countTest() {
+		SpannerOperations operations = mock(SpannerOperations.class);
+		new SpannerRepositoryImpl(operations, Object.class).count();
+		verify(operations, times(1)).count(eq(Object.class));
+	}
+
+	@Test
+	public void deleteByIdTest() {
+		SpannerOperations operations = mock(SpannerOperations.class);
+		String key = "key";
+		Object ret = new Object();
+		new SpannerRepositoryImpl(operations, Object.class).deleteById(key);
+		verify(operations, times(1)).delete(eq(Object.class), eq(Key.of(key)));
+	}
+
+	@Test
+	public void deleteTest() {
+		SpannerOperations operations = mock(SpannerOperations.class);
+		Object ob = new Object();
+		new SpannerRepositoryImpl(operations, Object.class).delete(ob);
+		verify(operations, times(1)).delete(eq(ob));
+	}
+
+	@Test
+	public void deleteManyObsTest() {
+		SpannerOperations operations = mock(SpannerOperations.class);
+		Iterable<String> obs = Arrays.asList("ob1", "ob2");
+		doAnswer(new Answer() {
```

Comment:
replace with lambda?

Reply:
@balopat changes all made from this pass!